src/VTableDumper.cpp

```diff
@@ -0,0 +1,216 @@
+        {
+            for (const auto& item : std::filesystem::directory_iterator(vtable_gen_output_function_bodies_path))
+            {
+                if (item.is_directory()) { continue; }
+                if (item.path().extension() != STR(".cpp")) { continue; }
+
+                File::delete_file(item.path());
+            }
+        }
+
+        if (std::filesystem::exists(vtable_gen_output_src_path))
+        {
+            for (const auto& item : std::filesystem::directory_iterator(vtable_gen_output_src_path))
+            {
+                if (item.is_directory()) { continue; }
+                if (item.path().extension() != STR(".cpp")) { continue; }
+
+                File::delete_file(item.path());
+            }
+        }
+
+        for (const auto& [class_name, class_entry] : type_container.get_class_entries())
+        {
+            Output::send(STR("Generating file '{}_VTableOffsets_{}_FunctionBody.cpp'\n"), pdb_name, class_entry.class_name_clean);
+            Output::Targets<Output::NewFileDevice> function_body_dumper;
+            auto& function_body_file_device = function_body_dumper.get_device<Output::NewFileDevice>();
+            function_body_file_device.set_file_name_and_path(vtable_gen_output_function_bodies_path / std::format(STR("{}_VTableOffsets_{}_FunctionBody.cpp"), pdb_name, class_name));
+            function_body_file_device.set_formatter([](File::StringViewType string) {
+                return File::StringType{ string };
+                });
+
+            for (const auto& [function_index, function_entry] : class_entry.functions)
+            {
+                auto local_class_name = class_entry.class_name;
+                if (auto pos = local_class_name.find(STR("Property")); pos != local_class_name.npos)
+                {
+                    local_class_name.replace(0, 1, STR("F"));
+                }
+
+                function_body_dumper.send(STR("if (auto it = {}::VTableLayoutMap.find(STR(\"{}\")); it == {}::VTableLayoutMap.end())\n"), local_class_name, function_entry.name, local_class_name);
+                function_body_dumper.send(STR("{\n"));
+                function_body_dumper.send(STR("    {}::VTableLayoutMap.emplace(STR(\"{}\"), 0x{:X});\n"), local_class_name, function_entry.name, function_entry.offset);
+                function_body_dumper.send(STR("}\n\n"));
+            }
+        }
+
+        auto template_file = std::format(STR("VTableLayout_{}_Template.ini"), pdb_name);
+        Output::send(STR("Generating file '{}'\n"), template_file);
+        Output::Targets<Output::NewFileDevice> ini_dumper;
+        auto& ini_file_device = ini_dumper.get_device<Output::NewFileDevice>();
+        ini_file_device.set_file_name_and_path(vtable_templates_output_path / template_file);
+        ini_file_device.set_formatter([](File::StringViewType string) {
+            return File::StringType{ string };
+            });
```

Comment:
```suggestion
        });
```
Poor indentation, not caused by mixed tabs & spaces, probably IDE giving bad suggestions.